src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,27 @@
+import { useContext } from 'react';
+import { TodosContext } from '../../context/TodosContext';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { Todo } from '../../types/Todo';
+import { Status } from '../../enums/Status';
+
+export const TodoList = () => {
+  const { todos, filter } = useContext(TodosContext);
+
+  let filteredTodos = todos;
```

Comment:
Instead of initializing `filteredTodos = todos`, you can define filteredTodos inside a useEffect or useMemo hook which will recompute todos when `todos` or `filter` changes. This will optimize your component to avoid unnecessary computations on each render.